jwst/pixel_replace/pixel_replace.py

```diff
@@ -326,6 +333,104 @@ def fit_profile(self, model):
+        # warning (despite well-defined behavior - return a NaN)
+        # so we suppress that here.
+        warnings.filterwarnings(action='ignore', message='All-NaN slice encountered')
+
+        log.info("Using minimum gradient method.")
+
+        # Input data, err, and dq values
+        indata = model.data
+        inerr = model.err
+        indq = model.dq
+
+        # Output data, err, and dq values
+        model_replaced = model.copy()
+        newdata = model_replaced.data
+        newerr = model_replaced.err
+        newdq = model_replaced.dq
+
+        # Make an array of x/y values on the detector
+        xsize, ysize = self.input.meta.subarray.xsize, self.input.meta.subarray.ysize
+        basex, basey = np.meshgrid(np.arange(xsize), np.arange(ysize))
+        pad = 1 # Padding around edge of array to ensure we don't look for neighbors outside array
+
+        # Find NaN-valued pixels
+        indx = np.where((np.isfinite(indata) == False)
+                        & (basey > pad) & (basey < ysize-pad) & (basex > pad) & (basex < xsize-pad))
+        # X and Y indices
+        yindx, xindx = indx[0], indx[1]
+
+        # Loop over these NaN-valued pixels
+        nreplaced = 0
+        for ii in range(0, len(xindx)):
+            left_data, right_data = indata[yindx[ii], xindx[ii] - 1], indata[yindx[ii], xindx[ii] + 1]
+            top_data, bottom_data = indata[yindx[ii] - 1, xindx[ii]], indata[yindx[ii] + 1, xindx[ii]]
+            left_err, right_err = inerr[yindx[ii], xindx[ii] - 1], inerr[yindx[ii], xindx[ii] + 1]
+            top_err, bottom_err = inerr[yindx[ii] - 1, xindx[ii]], inerr[yindx[ii] + 1, xindx[ii]]
+
+            # Compute absolute difference (slope) and average value in each direction (may be NaN)
+            diffs = np.array([np.abs(left_data - right_data), np.abs(top_data - bottom_data)])
+            interp_data = np.array([(left_data + right_data) / 2., (top_data + bottom_data) / 2.])
+            interp_err = np.array([(left_err + right_err) / 2., (top_err + bottom_err) / 2.])
+
+            # Replace with the value from the lowest absolute slope estimator that was not NaN
+            try:
+                indmin = np.nanargmin(diffs)
+                newdata[yindx[ii], xindx[ii]] = interp_data[indmin]
+                newerr[yindx[ii], xindx[ii]] = interp_err[indmin]
+                # If original pixel was in the science array set DQ accordingly
+                if ((indq[yindx[ii], xindx[ii]] & self.NON_SCIENCE) == 0):
+                    newdq[yindx[ii], xindx[ii]] = self.FLUX_ESTIMATED
+                # If original pixel was in non-science region set DQ accordingly
+                else:
+                    newdq[yindx[ii], xindx[ii]] = self.FLUX_ESTIMATED + self.NON_SCIENCE + self.DO_NOT_USE
+
+                nreplaced += 1
+            except Exception:
+                pass
+
+        log.info(f"Number of pixels replaced: {nreplaced}")
```

Comment:
I recall seeing similar log statements in the upper-level routine. Are we getting double reporting (or at least more than necessary)?